### ISSUE TITLE
python.buildEnv doesn't work with multiple outputs

### ISSUE THREAD

**Issue by domenkozar** [NixOS/nixpkgs#31857]
A http://sscce.org/:

```
$ nix-build -E 'with (import <nixpkgs> {}); python2.buildEnv.override { extraLibs = [ python2Packages.requests_toolbelt ]; }'     
...
building path(s) ‘/nix/store/pxma89rj5wf3p1lry578vxhhd8576410-python-2.7.14-env’
created 721 symlinks in user environment                                                                                          /nix/store/pxma89rj5wf3p1lry578vxhhd8576410-python-2.7.14-env

$ PYTHONPATH=/nix/store/pxma89rj5wf3p1lry578vxhhd8576410-python-2.7.14-env/lib/python2.7/site-packages/ python2 -c "import requests"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named requests
```

It works when requests is a direct dependency, but not when it's propagated.

I suspect it's multiple outputs fault:

```
$ nix-build -A pythonPackages.requests.all '<nixpkgs>'
/nix/store/xkj1nsa1wf6jzqgay5f7dsw523qzilag-python2.7-requests-2.18.4
/nix/store/bqk409lykpcsygh285ff8bfcmzyyvvq2-python2.7-requests-2.18.4-dev
```

```
$ nixos-version 
17.09.2023.57554d5ece (Hummingbird)
```

**Comment by FRidh** [comment]
The culprit is 28299f669adc41e5278372cad952fb1e1165b44b, see also https://github.com/NixOS/nixpkgs/commit/28299f669adc41e5278372cad952fb1e1165b44b#commitcomment-24598769.

```
$ cat /nix/store/cq0y7mcjcbz78b0w78ppfx2sw671n8fk-python3.6-requests-toolbelt-0.8.0/nix-support/propagated-native-build-inputs 
/nix/store/3yiag2s388zkvw1mi3mglj97bmjm2qv2-python3.6-requests-2.18.4-dev /nix/store/166s7l3yjqfc8dj5hfqjb09dbfvp1850-python3-3.6.3 /nix/store/hq27hydp4jnwa0kbccim6ghbd9l6h6q8-python3.6-setuptools-36.6.0 
```
And the `dev` output of `requests` does not depend on `out`.

The following solves your issue:
```
$ git diff
diff --git a/pkgs/development/interpreters/python/wrapper.nix b/pkgs/development/interpreters/python/wrapper.nix
index f42caf92c17..9bbff77d549 100644
--- a/pkgs/development/interpreters/python/wrapper.nix
+++ b/pkgs/development/interpreters/python/wrapper.nix
@@ -1,6 +1,6 @@
 { stdenv, python, buildEnv, makeWrapper
 , extraLibs ? []
-, extraOutputsToInstall ? []
+, extraOutputsToInstall ? [ "out" ]
 , postBuild ? ""
 , ignoreCollisions ? false }:
```

**Comment by domenkozar** [comment]
Not sure it's that commit, I'm running 17.09 and the commti is in master.

**Comment by FRidh** [comment]
Was cherry-picked as ea6947d6d25c0ec6310aa72097f741710ebdcd5a.

**Comment by FRidh** [comment]
That could have broken quite a lot of things, considering how much requests is used.

**Comment by FRidh** [comment]
Fixed in master and on 17.09 with 9d4e6f42e9a8e6cf5893950605bb0414d7e391ed.
